src/main/java/com/google/api/codegen/transformer/java/JavaGapicSurfaceTransformer.java

```diff
@@ -373,10 +375,20 @@ private StaticLangSettingsView generateSettingsClass(
     xsettingsClass.hasDefaultServiceAddress(interfaceConfig.hasDefaultServiceAddress());
     xsettingsClass.hasDefaultServiceScopes(interfaceConfig.hasDefaultServiceScopes());
     xsettingsClass.hasDefaultInstance(interfaceConfig.hasDefaultInstance());
+    xsettingsClass.generatorVersion(getGeneratorVersion());
 
     return xsettingsClass.build();
   }
 
+  private static String getGeneratorVersion() {
+    // TODO: Read version information from static file.
+    String version = JavaGapicSurfaceTransformer.class.getPackage().getImplementationVersion();
```

Comment:
Does this work when running through artman? It looks like it doesn't work in the baselines - maybe we need a versioning strategy like I introduced in gax?